File: LeetCode/Medium/3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        n,right,ans=len(nums),0,len(nums)
        for i in range(n):
            while right<n and nums[i]*k>=nums[right]:
                right+=1
            ans=min(ans,n-(right-i))
        return ans

File: LeetCode/Medium/3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.java
class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);
        int n=nums.length;
        int right = 0,ans=n;
        for(int i=0;i<n;i++){
            while(right<n && (long)nums[i]*k >= nums[right]) right++;
            ans=Math.min(ans,n-(right-i));
        }
        return ans;
    }
}